pelican/settings.py

```diff
@@ -212,6 +222,18 @@ def get_settings_from_file(path, default_settings=DEFAULT_CONFIG):
     return get_settings_from_module(module, default_settings=default_settings)
 
 
+def get_jinja_environment(settings):
+    """Sets the environment for Jinja"""
+
+    jinja_env = settings.get('JINJA_ENVIRONMENT', {})
```

Comment:
Slight issue. `.get` here doesn't do anything if the default is returned. You'll get an empty dict which is then populated but nothing will happen to the `settings`. If anything, `.setdefault` should be used instead.

Since any settings will use `DEFAULT_CONFIG` as base, there should be a `'JINJA_ENVIRONMENT'` key regardless, but still :).